contracts/crosschain/optimism/LibOptimism.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {ICrossDomainMessenger as Optimism_Bridge} from "../../vendor/optimism/ICrossDomainMessenger.sol";
+
+/**
+ * @dev Primitives for cross-chain aware contracts for
+ * [Optimism](https://www.optimism.io/).
+ */
+library LibOptimism {
+    /**
+     * @dev Returns whether the current function call is the result of a
+     * cross-chain message relayed by `bridge`.
+     */
+    function isCrossChain(address bridge) internal view returns (bool) {
+        return msg.sender == bridge;
+    }
+
+    /**
+     * @dev Returns the address of the sender that triggered the current
+     * cross-chain message through `bridge`.
+     *
+     * NOTE: {isCrossChain} should be checked before trying to recover the
+     * sender.
```

Comment:
I don't see how this is inconsistent. To me, it makes perfect sense to say that you should check if the call is cross-chain before doing something that might revert (not be recoverable).

Reply:
I see what you mean now. I interpreted this as saying that "crossChainSender _assumes_ that isCrossChain was called before". I think it should be reworded to say that it should be checked in order to avoid a potential revert.